engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1789,6 +1812,12 @@ namespace dmGameSystem
                 continue;
             }
 
+            float opacity = node_opacities[i];
+            if (opacity == 0) {
+                ++i;
+                continue;
+            }
```

Comment:
This is actually a bug!
If the node you're currently part of a different batch, we won't know since this check skips that check.

I suggest just removing this check here and let the corresponding RenderX function deal with it.

Reply:
True, that goes for the Bone ones as well.
In fairness, I don't think this code in general is super elegant since we need to process some redundant data.
But I'm not sure if it's worth performance wise to refactor.

Reply:
Isn't it a good thing? 
I mean if we break batch because of the node that has alpha == 0, in this case I'm checking that in advance and prevent batch breaking. But if I keep it only in RenderX functions, this node will break batch anyways

Reply:
But since it doesn't break batching, it might follow through to a RenderBox function, even though it's actually a Text node.

Sure, it might not be rendered, but I expect the Render*Type* functions to only receive nodes of the correct type.

Reply:
yeah, that's why we have the same checking in RenderType functions to skip this node before using any specific fields, isn't it?

Reply:
So, you think this possible win in draw-calls when we use the opacity value that's available for each RenderFunc to skip nodes of a wrong type before we operate with nodes (the same way we do for bones - we can't do something wrong if we do it after these checks), isn't worth it?

Reply:
The code might change later, and then it's very strange to have a function that processes e.g. TextNodes, to receive box nodes.